client/src/components/select-formula-view/SelectFormulaForm.js

```diff
@@ -16,64 +16,118 @@ import InputLabel from '@mui/material/InputLabel';
 import Select from '@mui/material/Select';
 import MenuItem from '@mui/material/MenuItem';
 import Typography from '@mui/material/Typography';
-import styled from 'styled-components';
+import StyledBox from './StyledBox';
 import ViewFormulaAccordion from './ViewFormulaAccordion';
+import AlertSnackbar from '../alerts/AlertSnackbar';
 
-const StyledBox = styled(Box)`
-  width: 53vw;
-  min-width:  400px;
-  max-width: 1000px;
-`;
 
-const SelectFormulaForm = ({ assignments, formulas, courseId }) => {
+const SelectFormulaForm = ({ assignments, formulas, navigateToCourseView, navigateToAttributeSelection }) => {
 
-  const [formula, setFormula] = useState('');
   const [codeSnippet, setCodeSnippet] = useState('');
-  const [selectedAssignments, setSelectedAssignments] = useState([]);
+  const [snackPack, setSnackPack] = useState([]);
+  const [alertOpen, setAlertOpen] = useState(false);
+  const [messageInfo, setMessageInfo] = useState(undefined);
+  
   const navigate = useNavigate();
 
+  const {
+    selectedAssignments, setSelectedAssignments,
+    selectedFormula, setSelectedFormula
+  } = useOutletContext();
+
+  useEffect(() => {
+    // set code snippet if user returns from attribute selection
+    if (selectedFormula.name !== undefined) {
+      setCodeSnippet(selectedFormula.codeSnippet);
+    }
+  }, []);
+
+  useEffect(() => {
+    // all attainments are checked at default -> add them to selected assignments
+    if (selectedAssignments.length === 0) {
+      setSelectedAssignments(assignments);
+    }
+  }, [assignments]);
+
+  // useEffect in charge of handling the back-to-back alerts
+  // makes the previous disappear before showing the new one
+  useEffect(() => {
+    if (snackPack.length && !messageInfo) {
+      setMessageInfo({ ...snackPack[0] });
+      setSnackPack((prev) => prev.slice(1));
+      setAlertOpen(true);
+    } else if (snackPack.length && messageInfo && alertOpen) {
+      setAlertOpen(false);
+    }
+  }, [snackPack, messageInfo, alertOpen]);
+
   const handleFormulaChange = (event) => {
     const newFormula = formulas.find(formula => formula.name == event.target.value);
     setCodeSnippet(newFormula.codeSnippet);
-    setFormula(event.target.value);
+    setSelectedFormula(newFormula);
   };
 
-  const handleSubmit = (event) => {
+  // checks that user has selected a function and at least one attainment
+  // if not, shows error message
```

Comment:
Should this error be handled as a message in the form? Alert could be used if the submitting goes wrong for some reason when the form is okay. Not a mandatory fix but just an idea.

Reply:
The error messages have been changed to be showed in the form